distribution/lib/Standard/Table/0.0.0-dev/src/Errors.enso

```diff
@@ -586,6 +586,25 @@ type Conversion_Failure
             Conversion_Failure.Text_Too_Long _ _ _ _ ->
                 rows_info + " have a text representation that does not fit the target type "+self.target_type.to_display_text+" when converting the column ["+self.related_column+"]."
 
+type Loss_Of_Integer_Precision
+    ## PRIVATE
+       Indicates that an automatic conversion of an integer column to a decimal
+       column is losing precision because some of the large integers cannot be
+       exactly represented by the `double` type.
+    Warning (affected_rows_count : Integer) (example_value : Integer) (example_value_converted : Decimal)
```

Comment:
Could this have a row number as well?

Reply:
Hm, I would have to modify the builder a bit to be able to know the row count. Currently there is not enough context.

But do you think it's useful? Tbh I'm not sure if the `row_number` that we have in a few of these errors is useful at all - hence I didn't think of adding it.

IMO the `example_value` serves this purpose enough - the user can see one of the values that are concerned, if the row number is needed they can find it by value. Row count is useful for sure to know if this affects like one row or like 50% of the rows.

I propose to see in practice - if we start encountering these and similar warnings in practice, let's see what we do with them in practical scenarios (e.g. bookclubs, other projects) and if such row number is helpful. My current gut feeling is that example values are more useful, because they illustrate e.g. that the number in question is very large. Or e.g. for date parsing failures, ideally our error should show an example value that failed to parse - this would allow the user to compare what it is with the format they specified and see what needs to be amended in the format to make it work. Row number does not help much here. (Ofc. it is more general as from row numbers we can figure out also the values - but that requires additional actions and IMO its better to immediately show examples).

Reply:
Actually, IMO what would be more useful here is `column_name` to know with which column this warning is associated, in a multi-column table.

However, it was not trivial how to get this information from the context so I abandoned this. I'm thinking it may be worth adding it as part of the next ticket #7514. But let me know if you think that I should add it now. I imagine it should not be very problematic to do so.

Reply:
Agreed; the value itself is enough to find it in the data; the row number would be nice but it's not worth a big effort.